tests/components/mystrom/test_init.py

```diff
@@ -0,0 +1,171 @@
+"""Test the myStrom init."""
+from unittest.mock import AsyncMock, PropertyMock, patch
+
+from pymystrom.exceptions import MyStromConnectionError
+
+from homeassistant.components.mystrom.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+DEVICE_NAME = "testmyStromdevice"
+
+ENTRY_ID = "uuid"
+DEVICE_MAC = "6001940376EB"
+
+
+async def init_integration(hass, device_type, bulb_type="strip"):
+    """Inititialize integration for testing."""
+    with patch(
+        "pymystrom.get_device_info",
+        side_effect=AsyncMock(return_value={"type": device_type, "mac": DEVICE_MAC}),
+    ), patch("pymystrom.switch.MyStromSwitch.get_state", return_value={}), patch(
+        "pymystrom.bulb.MyStromBulb.get_state", return_value={}
+    ), patch(
+        "pymystrom.bulb.MyStromBulb.bulb_type", bulb_type
+    ), patch(
+        "pymystrom.switch.MyStromSwitch.mac",
+        new_callable=PropertyMock,
+        return_value=DEVICE_MAC,
+    ), patch(
+        "pymystrom.bulb.MyStromBulb.mac",
+        new_callable=PropertyMock,
+        return_value=DEVICE_MAC,
+    ):
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id=ENTRY_ID,
+            unique_id="uuid",
+            data={CONF_HOST: "1.1.1.1"},
+            title=DEVICE_NAME,
+        )
+        config_entry.add_to_hass(hass)
```

Comment:
I would make this a common fixture in `conftest.py`:
```suggestion
@pytest.fixture
def config_entry(hass: HomeAssistant) -> MockConfigEntry:
    """Create and add a config entry."""
    config_entry = MockConfigEntry(
        domain=DOMAIN,
        entry_id=ENTRY_ID,
        unique_id="uuid",
        data={CONF_HOST: "1.1.1.1"},
        title=DEVICE_NAME,
    )
    config_entry.add_to_hass(hass)
    return config_entry
```

Reply:
done